opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -46,95 +62,179 @@ def __init__(
         if shutdown_on_exit:
             self._atexit_handler = register(self.shutdown)
 
-        self._metric_readers = []
-        self._metric_exporters = []
-        self._views = []
-        self._shutdown = False
+        self.__metric_readers = metric_readers
 
-    def get_meter(
-        self,
-        name: str,
-        version: Optional[str] = None,
-        schema_url: Optional[str] = None,
-    ) -> Meter:
+        for metric_reader in self._metric_readers:
+            metric_reader._register_meter_provider(self)
 
-        if self._shutdown:
-            _logger.warning(
-                "A shutdown `MeterProvider` can not provide a `Meter`"
-            )
-            return _DefaultMeter(name, version=version, schema_url=schema_url)
+        self.__metric_exporters = metric_exporters
 
-        return Meter(InstrumentationInfo(name, version, schema_url), self)
+        self.__views = views
 
-    def shutdown(self):
-        # FIXME implement a timeout
+        if self.__use_always_matching_view:
+            self.__views = [*self.__views, View(instrument_name="*")]
 
-        if self._shutdown:
-            _logger.warning("shutdown can only be called once")
-            return False
+        self.__resource = resource
+        self._shutdown = False
+        self.__synchronous_instruments = []
+        self.__asynchronous_instruments = []
 
-        result = True
+    @property
+    def _metric_readers(self):
+        return self.__metric_readers
 
-        for metric_reader in self._metric_readers:
-            result = result and metric_reader.shutdown()
+    @property
+    def _use_always_matching_view(self):
+        return self.__use_always_matching_view
 
-        for metric_exporter in self._metric_exporters:
-            result = result and metric_exporter.shutdown()
+    @property
+    def _resource(self):
+        return self.__resource
 
-        self._shutdown = True
+    @property
+    def _metric_exporters(self):
+        return self.__metric_exporters
 
-        if self._atexit_handler is not None:
-            unregister(self._atexit_handler)
-            self._atexit_handler = None
+    @property
+    def _views(self):
+        return self.__views
 
-        return result
+    @property
+    def _synchronous_instruments(self):
+        return self.__synchronous_instruments
+
+    @property
+    def _asynchronous_instruments(self):
+        return self.__asynchronous_instruments
+
+    def _create_instrument(
+        self, instrument_type, instrumentation_info, name, unit, description
+    ):
+
+        instrument = instrument_type(
+            instrumentation_info, name, unit=unit, description=description
+        )
+
+        with self._lock:
+            if isinstance(instrument, Synchronous):
+                self.__synchronous_instruments.append(instrument)
+
+            else:
+                self.__asynchronous_instruments.append(instrument)
+
+        return instrument
+
+    def _create_counter(
```

Comment:
Why are these `create_*()` methods on the `MeterProvider` class? Why can't this be put on the `Meter`?

Reply:
They are in the `MeterProvider` because of the need to register Asynchronous instruments, but since you prefer them not being there I have removed them :+1: